code/modules/antagonists/heretic/items/heretic_blades.dm

```diff
@@ -137,6 +142,53 @@
 	icon_state = "dark_blade"
 	inhand_icon_state = "dark_blade"
 	after_use_message = "The Torn Champion hears your call..."
+	///If our blade is currently infused with the mansus grasp
+	var/infused = FALSE
+
+/obj/item/melee/sickly_blade/dark/afterattack(atom/target, mob/user, click_parameters)
+	. = ..()
+	if(!infused || target == user)
+		return
+	var/datum/antagonist/heretic/heretic_datum = IS_HERETIC(user)
+	if(!heretic_datum)
+		return
+
+	//Apply our heretic mark
+	var/datum/heretic_knowledge/mark/blade_mark/mark_to_apply = heretic_datum.get_knowledge(/datum/heretic_knowledge/mark/blade_mark)
+	if(!mark_to_apply)
+		return
+	mark_to_apply.create_mark(user, target)
+
+	//Remove the infusion from any blades we own (and update their sprite)
+	for(var/obj/item/melee/sickly_blade/dark/to_infuse in user.get_all_contents_type(/obj/item/melee/sickly_blade/dark))
+		to_infuse.infused = FALSE
+		to_infuse.update_appearance(UPDATE_ICON)
+	user.update_held_items()
+
+	if(!check_behind(user, target) || !isliving(target))
```

Comment:
Shouldn't this isliving be wayyy earlier, like the second line of this proc? 

I don't think you can apply a mansus grasp mark to a non-living mob